### DISCUSSION TITLE
How do I get preset remark-preset-lint-markdown-style-guide to ignore frontmatter?

### DISCUSSION THREAD

**Discussion by arrrgi** [remarkjs/remark#645] [Q&A] [answered]
My Markdown has YAML frontmatter. eg:

```markdown
---
id: doc2
title: Document Number 2
---

This is a link to [another document.](doc3.md) This is a link to an [external page.](http://www.example.com/)
```

When linting with both the `remark-frontmatter` and `remark-preset-lint-markdown-style-guide` plugins I get a warning that fields in the frontmatter don't match the **"header: atx"** rule.

Am I doing something wrong or should I report this as a bug?

My basic Remark config looks like this:

```javascript
exports.plugins = [
  'remark-frontmatter',
  'remark-external-links',
  'remark-capitalize',
  'remark-preset-lint-markdown-style-guide'
];

exports.settings = {
  strong: '*'
};
```

**Comment by wooorm** [comment]
That should indeed work. Can you make a reproduction for this?

**Reply by arrrgi** [reply]
Mocked up a small reproduction for this here --> https://github.com/madcatsu/remark-lint-override-issue

**Reply by wooorm** [reply]
Thanks!

Ah. Well, you’re using eslint and/or eslint-plugin-mdx, that’s the issue.

remark supports this (`npx remark-cli doc1.md` works), but the rest of the tooling fails.

**Reply by arrrgi** [reply]
Well spotted.

Let's say my actual project is using `lint-staged`, what would be the most efficient way for linting both MDX and Markdown? My current config does this with ESLint and I'd like to handle both types if possible.

**Reply by wooorm** [reply]
They’re two different data formats. While using similar tools, what you write in either is incompatible with the other.

You could use remark-cli for the markdown. Or alternatively I think there’s also an eslint-markdown project?
And then use eslint-mdx for the MDX

**Reply by arrrgi** [reply]
This looks like frontmatter support is meant to work in eslint-mdx, see here: https://github.com/mdx-js/eslint-mdx/issues/147

I understand your point above but this particular fix looks like it was intended to resolve this particular problem, I cannot get it to work though even when exporting remark-frontmatter as @JounQin suggests

Edit: happy to handle linting the two file types separately, but it looks like it's not detected correctly in MDX

**Reply by arrrgi** [reply]
What's also interesting is that I can lint an MDX with frontmatter on this other reproduction repo: https://github.com/madcatsu/eslint-mdx-frontmatter-issue

Granted some packages in there are outdated but this was a working example of ESLint working on MDX files with frontmatter.

**Reply by arrrgi** [reply]
After testing ESLint config and version between both repos I noticed that I can use eslint-mdx on a file target just fine without it reporting errors.

Comparatively, if I have ESLint walk the file tree and look for .mdx files, it then gives the warning that frontmatter should be **headers: atx**

**Reply by wooorm** [reply]
Ah, then it’s this one: https://github.com/remarkjs/remark-frontmatter#important.

mdx-js/mdx is still on an old remark unfortunately, so downgrading should do the trick

**Reply by JounQin** [reply]
I would recommend you to use `prettier` for formatting instead.

[`remark-preset-prettier`](https://github.com/JounQin/remark-preset-prettier) may help.

**Reply by arrrgi** [reply]
Thanks @JounQin - **Prettier** is already inside my ecosystem. Linting is an additional requirement to keep syntax the same across MDX and MD in my use case

**Reply by JounQin** [reply]
`remark-lint-heading-style` may be conflicted with `prettier`, so just use `remark-preset-prettier`, there will be no related issue including the current one you're facing.

**Reply by arrrgi** [reply]
I see how this is useful as it will just ignore the header: atx rule requirement then

**Comment by JounQin** [comment, marked as answer]
`remark-mdx@1` is not compatible with `remark@13` yet, but `remark-cli` and `remark-lint` have been upgraded to `remark@13.0`.

What means when you're running `eslint` with `eslint-mdx`, `remark@12` is used, but when you're running `remark-cli`, `remark@13` is used.

@wooorm I'm also wondering can we support `remark@13` for `remark-mdx@1` at the same time, since `remark-mdx@2` is not ready yet.

![image](https://user-images.githubusercontent.com/8336744/109652514-dabe0b80-7b9a-11eb-9b0f-b251877239f4.png)

**Reply by wooorm** [reply]
remark@13 is a 100% different parser. That’s why it’s not possible to combine all these things, and it was major versions through the whole ecosystem!

**Reply by arrrgi** [reply]
Not quite back to square one but at least it's been identified what was happening, thanks both. Sounds like I should sit tight till the both remark-mdx and eslint-mdx capture the remark@13 changes.
